### ISSUE TITLE
Sorting entries in helm-org-ql

### ISSUE THREAD

**Issue by akirak** [alphapapa/org-ql#101]
I am using `helm-org-ql-source` in [this project](https://github.com/akirak/org-multi-wiki), and now I'd rather want to display items in a way where lower-level items are shown first. Is it possible to sort entries in `helm-org-ql`?

**Comment by alphapapa** [comment]
An argument could easily be added to `helm-org-ql-source` to pass a sorting argument to `org-ql-select`.  https://github.com/alphapapa/org-ql/blob/06481960764a55a36d886e1db79a58258d5d2ffb/helm-org-ql.el#L188

**Comment by akirak** [comment]
I added `:sort` argument to `org-ql-select` in `helm-org-ql.el`, but it didn't work. It must be because the action returns a heading text, not an element.

**Comment by alphapapa** [comment]
Oops, you're right, sorry for sending you down the wrong path there.

I guess what we should do is, when a sorting argument is given, use the `element` action, and then map the `helm-org-ql--heading ` across the results (which will require either going to each result's buffer and position).  We could do that regardless of whether there's a sorting argument, but I want the Helm command to be as fast and responsive as possible, so I'd prefer to not go through that extra work when sorting isn't used.

**Comment by akirak** [comment]
Thank you for your comment.

I also wonder if it is possible to skip subtrees (#https://github.com/alphapapa/org-ql/pull/89) in `helm-org-ql`. It probably should be optional, but it will have a great impact on sorting performance, since it will definitely reduce the number of items.

**Comment by alphapapa** [comment]
It would have to be optional, yes.

**Comment by akirak** [comment]
Maybe you can parse elements and generate outlines in a single path. With `candidate-transformer`, `filtered-candidate-transformer`, or `filter-one-by-one` of the helm source, you can store both the headline and the element in a single cell. Parsing elements would be a performance cost anyway, so you would want to avoid it when sorting is unused.

`helm-source` is indeed an interesting read. You may be able to change the sorting method after dispatching a helm session using `candidate-transformer`.

**Comment by alphapapa** [comment]
Yeah, there are a lot of possibilities.  I generally try not to get too fancy with Helm sources' options.  I used transformers for sorting in org-rifle, but something changed and it broke (not sure whether due to a change in Helm, Org, or my own code).  Sometimes Helm changes without warning, as well, and I try to minimize chances of breakage due to something that's out of my control.
